doc/rtd/explanation/format.rst

```diff
@@ -24,64 +71,121 @@ These things may include:
 - importing certain SSH keys or host keys
 - *and many more...*
 
-See the :ref:`yaml_examples` section for a commented set of examples of
-supported cloud config formats.
-
-Begins with: ``#cloud-config`` or ``Content-Type: text/cloud-config`` when
-using a MIME archive.
+Many modules are available to process cloud-config data. These modules
+may run once per instance, every boot, or once ever. See the associated
+module to determine the run frequency.
 
-.. note::
-   Cloud config data can also render cloud instance metadata variables using
-   :ref:`jinja templates <instancedata-Using>`.
+See the :ref:`yaml_examples` section for a set of commented examples of
+supported cloud config formats.
 
 .. _user_data_script:
 
 User data script
 ================
 
-Typically used by those who just want to execute a shell script.
+| **Header:** #!
+| **Content-Type:** text/x-shellscript
+
+Example
+-------
+
+.. code-block:: shell
 
-Begins with: ``#!`` or ``Content-Type: text/x-shellscript`` when using a MIME
-archive.
+    #!/bin/sh
+    echo "Hello World" > /var/tmp/output.txt
 
-User data scripts can optionally render cloud instance metadata variables using
-:ref:`jinja templates <instancedata-Using>`.
+Explanation
+-----------
 
-Example script
---------------
+A user data script is a single shell script to be executed once per instance.
+User data scripts are run relatively late in the boot process, during the
+'modules:final' stage as part of the "cc_scripts_user" module.
```

Comment:
I know that `modules:final" text that is commonly seen in the log, but I think it would be easier to standardize on the more generic "final" stage, since the "modules" entrypoint is really just an implementation detail that the user shouldn't need to know about.

Also, could we link to the respective stages on the boot order page both here and in the cloud boothook section?